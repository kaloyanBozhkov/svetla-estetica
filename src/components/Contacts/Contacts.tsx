import type { ReactElement } from 'react'
import styles from './contacts.module.scss'

const phones = [
        ['Cellulare', '(+39)', '393 5026 350'],
        ['Fisso', '(+39)', '35 0632279'],
    ],
    social = [
        {
            label: 'Svetla Estetica Dalmine',
            className: styles.fb,
            link: 'https://www.facebook.com/SvetlaEsteticaDalmine',
        },
        {
            label: '@svetlaesteticadalmine',
            className: styles.ig,
            link: 'https://www.instagram.com/svetlaesteticadalmine/',
        },
        {
            label: '(+39) 393 5026 350',
            className: styles.ws,
            link: 'https://wa.me/+393935026350',
        },
    ],
    Contacts = (): ReactElement => (
        <div className={styles.contacts}>
            <div>
                {/* <p>Telefono</p> */}
                {phones.map(([type, ext, number]) => (
                    <p key={type}>
                        <span>{type}</span>
                        <a href={`tel:${`${ext} ${number}`.replace(/[(,), ]/g, '')}`}>
                            <span>{ext}</span>
                            <span>{number}</span>
                        </a>
                    </p>
                ))}
            </div>
            <div>
                {social.map(({ className, label, link }) => (
                    <a key={className} href={link}>
                        <div className={className}>
                            <p>{label}</p>
                        </div>
                    </a>
                ))}
            </div>
        </div>
    )

export default Contacts
