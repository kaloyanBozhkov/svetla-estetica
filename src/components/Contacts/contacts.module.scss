@import "scss/mixins.scss";
@import "scss/variables.scss";

.contacts {
    width: 100%;
    display: flex;
    gap: 24rem;

    @include sidePaddings;
    padding-top: 4rem;
    padding-bottom: 4rem;

    position: relative;
    &::before {
        content: "";
        width: 1px;
        position: absolute;
        left: 50%;
        transform: translateX(-50%);
        background-color: $lighterGrey;
        top: 20%;
        bottom: 20%;
    }

    > div {
        flex: 1;
        display: flex;
        flex-direction: column;
        justify-content: center;
        gap: 2rem;

        > p {
            font-weight: lighter;
            font-size: 20px;
            display: flex;
            justify-content: space-between;
        }

        &:first-child {
            > p {
                &:not(:first-child) {
                    &:not(:last-child) {
                        margin-bottom: 1rem;
                    }
                }
            }
        }

        &:last-child {
            .fb,
            .ig,
            .ws {
                display: flex;
                align-items: center;
                gap: 3rem;

                p {
                    font-size: 1.6rem;
                    @include breakpoint(tablet-portrait) {
                        font-size: 3rem;
                    }
                }

                &::before {
                    content: "";
                    width: 2.8rem;
                    height: 2.8rem;
                    background-size: cover;
                }
            }

            .fb {
                p {
                    color: #17a8fd;
                }

                &::before {
                    background-image: url("/assets/icons/fb.svg");
                }
            }

            .ig {
                p {
                    color: #c03b96;
                }

                &::before {
                    background-image: url("/assets/icons/ig.svg");
                }
            }

            .ws {
                p {
                    color: #3fc350;
                }

                &::before {
                    background-image: url("/assets/icons/ws.svg");
                }
            }
        }
    }

    @include breakpoint(tablet-portrait) {
        padding-top: 5rem;
        padding-bottom: 5rem;
        flex-direction: column;
        gap: 8rem;

        &::before {
            display: none;
        }
    }
}
