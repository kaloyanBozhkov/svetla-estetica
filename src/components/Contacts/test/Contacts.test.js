import { shallow } from 'enzyme'
import React from 'react'
import Contacts from '../Contacts'

describe('testing Contacts', () => {
    it('should render component', () => {
        expect(shallow(<Contacts />)).toMatchSnapshot()
    })
})
