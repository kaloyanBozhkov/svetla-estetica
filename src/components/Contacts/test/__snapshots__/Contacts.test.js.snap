// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Contacts should render component 1`] = `
<div
  className="contacts"
>
  <div>
    <p
      key="Cellulare"
    >
      <span>
        Cellulare
      </span>
      <a
        href="tel:+393935026350"
      >
        <span>
          (+39) 393 5026 350
        </span>
      </a>
    </p>
    <p
      key="Fisso"
    >
      <span>
        Fisso
      </span>
      <a
        href="tel:+39350632279"
      >
        <span>
          (+39) 35 0632279
        </span>
      </a>
    </p>
  </div>
  <div>
    <a
      href="https://www.facebook.com/SvetlaEsteticaDalmine"
      key="fb"
    >
      <div
        className="fb"
      >
        <p>
          Svetla Estetica Dalmine
        </p>
      </div>
    </a>
    <a
      href="https://www.instagram.com/svetlaesteticadalmine/"
      key="ig"
    >
      <div
        className="ig"
      >
        <p>
          @svetlaesteticadalmine
        </p>
      </div>
    </a>
    <a
      href="https://wa.me/+393935026350"
      key="ws"
    >
      <div
        className="ws"
      >
        <p>
          (+39) 393 5026 350
        </p>
      </div>
    </a>
  </div>
</div>
`;
